master/internal/trial.go

```diff
@@ -296,6 +311,14 @@ func (t *trial) Receive(ctx *actor.Context) error {
 		// the code below this switch statement to handle releasing resources in
 		// the scheduler. This should be refactored into the terminating logic.
 
+	case trialWatchPreemption:
+		t.preemptionWatchers[msg.signal] = true
+		if t.PendingGracefulTermination {
+			msg.signal <- struct{}{}
+		}
+	case trialUnwatchPreemption:
+		delete(t.preemptionWatchers, msg.signal)
```

Comment:
Do you not have to close channels in go?  I'm actually unsure.

Reply:
No, channels are gc'd when all references too them are gone